### PR TITLE
feat(cdk): add `TuiAlertContext`

### LINKED ISSUE #1692: 📚 - `TuiAlertService` nothing found
### What is the affected URL?

https://taiga-ui.dev/getting-started

### Description

image

### Which browsers have you used?

- [X] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Which operating systems have you used?

- [X] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [x] Documentation content changes

## What is the current behavior?

Closes #1692 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/demo/src/modules/services/alerts/examples/import/module.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+```ts
+import {TuiNotificationsModule, TuiRootModule} from '@taiga-ui/core';
+//...
+
+@NgModule({
+  imports: [
+    TuiRootModule,
+    TuiNotificationsModule,
```
Comment (nsbarsukov) [inline projects/demo/src/modules/services/alerts/examples/import/module.md:8]:
```suggestion
    TuiAlertModule,
```

File: projects/demo/src/modules/services/alerts/examples/import/module.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+```ts
+import {TuiNotificationsModule, TuiRootModule} from '@taiga-ui/core';
```
Comment (nsbarsukov) [inline projects/demo/src/modules/services/alerts/examples/import/module.md:2]:
```suggestion
import {TuiAlertModule, TuiRootModule} from '@taiga-ui/core';
```

File: projects/demo/src/modules/services/alerts/examples/import/service-usage-component.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+```ts
+import {PolymorpheusComponent} from '@tinkoff/ng-polymorpheus';
+import {TuiNotificationsService} from '@taiga-ui/core';
```
Comment (nsbarsukov) [inline projects/demo/src/modules/services/alerts/examples/import/service-usage-component.md:3]:
```suggestion
import {TuiAlertService} from '@taiga-ui/core';
```

File: projects/demo/src/modules/services/alerts/examples/import/service-usage-component.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import {CustomNotificationComponent} from 'customNotification.component';
+//...
+
+export class MyComponent {
+  constructor(@Inject(TuiNotificationsService) private readonly notificationsService: TuiNotificationsService) {
```
Comment (nsbarsukov) [inline projects/demo/src/modules/services/alerts/examples/import/service-usage-component.md:8]:
```suggestion
  constructor(@Inject(TuiAlertService) private readonly alertService: TuiAlertService) {
```

File: projects/demo/src/modules/services/alerts/examples/import/service-usage.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
+```ts
+import {TuiNotificationsService} from '@taiga-ui/core';
```
Comment (nsbarsukov) [inline projects/demo/src/modules/services/alerts/examples/import/service-usage.md:2]:
```suggestion
import {TuiAlertService} from '@taiga-ui/core';
```

File: projects/demo/src/modules/services/alerts/examples/2/index.ts [hunk 6]
```diff
@@ -38,8 +38,8 @@ export class TuiNotificationsExampleComponent2 {
     }
 
     showDepositAlert(): void {
-        this.notificationsService
-            .show(this.depositTemplate || '', {
+        this.alertService
+            .open(this.depositTemplate || ('' as any), {
```
Comment (nsbarsukov) [inline projects/demo/src/modules/services/alerts/examples/2/index.ts:42]: can we add TODO to remove it later?

looks terrible(

**Review by nsbarsukov — APPROVED** [review]

File: projects/demo/src/modules/services/alerts/examples/2/index.ts (hunk 6 above)
Comment (splincode) [reply projects/demo/src/modules/services/alerts/examples/2/index.ts:42]: Of course, I think this is a problem that will need to be solved later

Comment (waterplea) [reply projects/demo/src/modules/services/alerts/examples/2/index.ts:42]: Why string does not pass as a valid argument?

File: projects/demo/src/modules/services/alerts/examples/3/index.ts [hunk 7]
```diff
@@ -34,10 +34,10 @@ export class TuiNotificationsExampleComponent3 {
             )
             .pipe(
                 switchMap(response =>
-                    notificationsService.show(`Got a value — ${response}`, {
+                    alertService.open(`Got a value — ${response}`, {
                         label: 'Information',
                     }),
-                ),
+                ) as any,
```
Comment (waterplea) [inline projects/demo/src/modules/services/alerts/examples/3/index.ts:40]: Why?

File: projects/demo/src/modules/services/alerts/examples/4/index.ts [hunk 8]
```diff
@@ -35,10 +35,10 @@ export class TuiNotificationsExampleComponent4 {
             )
             .pipe(
                 switchMap(response =>
-                    notificationsService.show(`Got a value — ${response}`, {
+                    notificationsService.open(`Got a value — ${response}`, {
                         label: 'Information',
                     }),
-                ),
+                ) as any,
```
Comment (waterplea) [inline projects/demo/src/modules/services/alerts/examples/4/index.ts:41]: Here as well

File: projects/demo/src/modules/services/alerts/examples/import/module.md (hunk 1 above)
Comment (splincode) [reply projects/demo/src/modules/services/alerts/examples/import/module.md]: fixed

File: projects/demo/src/modules/services/alerts/examples/import/module.md (hunk 2 above)
Comment (splincode) [reply projects/demo/src/modules/services/alerts/examples/import/module.md]: fixed

File: projects/demo/src/modules/services/alerts/examples/import/service-usage-component.md (hunk 4 above)
Comment (splincode) [reply projects/demo/src/modules/services/alerts/examples/import/service-usage-component.md]: fixed

File: projects/demo/src/modules/services/alerts/examples/import/service-usage-component.md (hunk 3 above)
Comment (splincode) [reply projects/demo/src/modules/services/alerts/examples/import/service-usage-component.md]: fixed

File: projects/demo/src/modules/services/alerts/examples/import/service-usage.md (hunk 5 above)
Comment (splincode) [reply projects/demo/src/modules/services/alerts/examples/import/service-usage.md]: fixed

File: projects/demo/src/modules/services/alerts/examples/2/index.ts (hunk 6 above)
Comment (splincode) [reply projects/demo/src/modules/services/alerts/examples/2/index.ts]: fixed

File: projects/demo/src/modules/services/alerts/examples/4/index.ts (hunk 8 above)
Comment (splincode) [reply projects/demo/src/modules/services/alerts/examples/4/index.ts]: fixed

File: projects/demo/src/modules/services/alerts/examples/3/index.ts (hunk 7 above)
Comment (splincode) [reply projects/demo/src/modules/services/alerts/examples/3/index.ts]: fixed

File: projects/demo/src/modules/services/alerts/examples/4/index.ts [hunk 9]
```diff
@@ -2,29 +2,29 @@ import {Component, Inject, Injector} from '@angular/core';
...
     changeDetection,
     encapsulation,
 })
-export class TuiNotificationsExampleComponent4 {
+export class TuiAlertsExampleComponent4 {
     readonly notification: Observable<void>;
 
     constructor(
-        @Inject(TuiNotificationsService) notificationsService: TuiNotificationsService,
+        @Inject(TuiAlertService) notificationsService: TuiAlertService,
```
Comment (waterplea) [inline projects/demo/src/modules/services/alerts/examples/4/index.ts:22]: Let's rename too

Comment (splincode) [reply projects/demo/src/modules/services/alerts/examples/4/index.ts]: fixed

**Comment by splincode** [comment]
@waterplea @nsbarsukov please, review me)

File: projects/demo/src/modules/services/alerts/examples/2/index.ts [hunk 10]
```diff
@@ -1,35 +1,32 @@
 import {Component, Inject, TemplateRef, ViewChild} from '@angular/core';
 import {changeDetection} from '@demo/emulate/change-detection';
 import {encapsulation} from '@demo/emulate/encapsulation';
-import {
-    TuiNotification,
-    TuiNotificationContentContext,
-    TuiNotificationsService,
-} from '@taiga-ui/core';
+import {TuiAlertContext} from '@taiga-ui/cdk/interfaces';
```
Comment (nsbarsukov) [inline projects/demo/src/modules/services/alerts/examples/2/index.ts:4]:
```suggestion
import {TuiAlertContext} from '@taiga-ui/cdk';
```

**Review by nsbarsukov — APPROVED** [review]

File: projects/demo/src/modules/services/alerts/examples/2/index.ts (hunk 10 above)
Comment (splincode) [reply projects/demo/src/modules/services/alerts/examples/2/index.ts]: fixed

File: projects/cdk/abstract/dialog.service.ts [hunk 11]
```diff
@@ -23,8 +23,8 @@ export abstract class AbstractTuiDialogService<T> extends Observable<
         super(observer => this.dialogs$.subscribe(observer));
     }
 
-    open<G>(
-        content: PolymorpheusContent<TuiBaseDialogContext<G> & T>,
+    open<G = void>(
+        content: PolymorpheusContent<TuiAlertContext<T, G>>,
```
Comment (waterplea) [inline projects/cdk/abstract/dialog.service.ts:27]: This is a common abstract class for both dialogs and alerts, please revert this interface change.

Comment (splincode) [reply projects/cdk/abstract/dialog.service.ts]: fixed

**Pull request merged**